payroll/models/hr_salary_rule.py

```diff
@@ -190,25 +190,23 @@ def _compute_rule(self, localdict):
         self.ensure_one()
         if self.amount_select == "fix":
             try:
-                return (
-                    self.amount_fix,
-                    float(safe_eval(self.quantity, localdict)),
-                    100.0,
-                    False,  # result_name is always False if not computed by python.
-                )
+                return {
+                    "quantity": float(safe_eval(self.quantity, localdict)),
+                    "rate": 100.0,
+                    "amount": self.amount_fix,
+                }
```

Comment:
I think that if we are going to extract the logic of "python" compute mode, we should do the same with each possible value of amount_select. 

We could have tree functions: 
- _compute_values_python
- _compute_values_fixed
- _compute_values_percentage

This is useful if the user want to inherit a specific part of the function, and also useful if in the future we add new calculation modes.